### ISSUE TITLE
Could not find CFPropertyList-2.3.6 in any of the sources / Homebrew install not working

### ISSUE THREAD

**Issue by saruye** [fastlane/fastlane#12118]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I installed fastlane from brew (`brew cask install fastlane`) and when I tried to run fastlane it complained about not finding  CFPropertyList in the right version. It is asking for version 2.3.6 but version 2.3.5 is included in the package that can be downloaded from fastlane (https://fastlane.tools/fastlane.zip) 

The same error came when just doing a `fastlane --version` so I don't think it is related to my configuration

I had fastlane installed before through gem. 

##### Complete output when running fastlane, including the stack trace and command used
<details>
[me:...e/ios/app-ios/app ios]$ fastlane staging --capture_output                                                                                                                                                                                            (master✱)
[11:31:33]: fastlane detected a Gemfile in the current directory
[11:31:33]: however it seems like you don't use `bundle exec`
[11:31:33]: to launch fastlane faster, please use
[11:31:33]:
[11:31:33]: $ bundle exec fastlane staging --capture_output
[11:31:33]:
[11:31:33]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
DEBUG [2018-03-21 11:31:33.96]: Checking if there are any plugins that should be loaded...
INFO [2018-03-21 11:31:35.13]: ---------------------------
INFO [2018-03-21 11:31:35.13]: --- Step: opt_out_usage ---
INFO [2018-03-21 11:31:35.13]: ---------------------------
INFO [2018-03-21 11:31:35.13]: Disabled upload of used actions
INFO [2018-03-21 11:31:35.13]: ----------------------------------------
INFO [2018-03-21 11:31:35.13]: --- Step: Verifying fastlane version ---
INFO [2018-03-21 11:31:35.13]: ----------------------------------------
INFO [2018-03-21 11:31:35.13]: Your fastlane version 2.86.1 matches the minimum requirement of 1.55.0  ✅
INFO [2018-03-21 11:31:35.14]: ------------------------------
INFO [2018-03-21 11:31:35.14]: --- Step: default_platform ---
INFO [2018-03-21 11:31:35.14]: ------------------------------
INFO [2018-03-21 11:31:35.14]: Driving the lane 'ios staging' 🚀
INFO [2018-03-21 11:31:35.14]: -----------------------
INFO [2018-03-21 11:31:35.14]: --- Step: cocoapods ---
INFO [2018-03-21 11:31:35.14]: -----------------------
INFO [2018-03-21 11:31:35.20]: gem 'cocoapods' is already installed
INFO [2018-03-21 11:31:35.20]: $ bundle exec pod install --repo-update
INFO [2018-03-21 11:31:35.72]: ▸ bundler: failed to load command: pod (/Users/me/.fastlane/bin/bundle/bin/pod)
INFO [2018-03-21 11:31:35.72]: ▸ Bundler::GemNotFound: Could not find CFPropertyList-2.3.6 in any of the sources
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/spec_set.rb:87:in `block in materialize'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/spec_set.rb:80:in `map!'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/spec_set.rb:80:in `materialize'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/definition.rb:176:in `specs'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/definition.rb:235:in `specs_for'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/definition.rb:224:in `requested_specs'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/runtime.rb:118:in `block in definition_method'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/runtime.rb:19:in `setup'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler.rb:100:in `setup'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/setup.rb:20:in `<top (required)>'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:55:in `require'
INFO [2018-03-21 11:31:35.72]: ▸ /Users/me/.fastlane/bin/bundle/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:55:in `require'
ERROR [2018-03-21 11:31:35.74]: Exit status of command 'bundle exec pod install --repo-update' was 1 instead of 0.
bundler: failed to load command: pod (/Users/me/.fastlane/bin/bundle/bin/pod)
Bundler::GemNotFound: Could not find CFPropertyList-2.3.6 in any of the sources
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/spec_set.rb:87:in `block in materialize'
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/spec_set.rb:80:in `map!'
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/spec_set.rb:80:in `materialize'
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/definition.rb:176:in `specs'
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/definition.rb:235:in `specs_for'
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/definition.rb:224:in `requested_specs'
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/runtime.rb:118:in `block in definition_method'
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/runtime.rb:19:in `setup'
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler.rb:100:in `setup'
  /Users/me/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/setup.rb:20:in `<top (required)>'
  /Users/me/.fastlane/bin/bundle/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:55:in `require'
  /Users/me/.fastlane/bin/bundle/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:55:in `require'

INFO [2018-03-21 11:31:36.53]: Successfully generated documentation at path '/Users/me/workspace/ios/app-ios/app ios/fastlane/README.md'
ERROR [2018-03-21 11:31:36.55]: fastlane finished with errors
ERROR [2018-03-21 11:31:37.20]: Error finding relevant GitHub issues: undefined method `map' for nil:NilClass
</details>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.3                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.14.3 (Apple Git-98)           |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.13.3 (17D102)                   |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.86.1  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.2        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.6        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.1.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| naturally                 | 2.1.0        |
| simctl                    | 1.6.2        |
| uber                      | 0.0.15       |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| libxml-ruby               | 3.0.0        |
</details>

*generated on:* **2018-03-21**
</details>

**Comment by joshdholtz** [comment]
@saruye Can you paste what your `Fastfile` looks like? Thanks!

**Comment by saruye** [comment]
@joshdholtz 

here is the file content. Maybe its related to cocoapods? they seem to install first. after that its not working anymore. But what irritated me, that the problem is globally. Even running `fastlane --version` anywhere doesn't work.

<details>

```
fastlane_version "2.86.0"

default_platform :ios

platform :ios do
  before_all do
    ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    cocoapods(
      repo_update: true
    )
  end

  private_lane :build do |options| 
    appId = CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier)

    update_info_plist(
      app_identifier: appId,
      plist_path: "app-Info.plist"
    )
    update_app_identifier(
      app_identifier: appId,
      plist_path: "app-Info.plist"
    )
    
    scan(
        scheme: "app",
        output_types: 'junit',
        custom_report_file_name: 'report.xml'
    ) 

    increment_build_number({
      build_number: latest_testflight_build_number.to_i + 1
    })
    match(type: "appstore", readonly: true)

    if options.key?(:family) 
      gym(scheme: "scheme",
  	  xcargs: "TARGETED_DEVICE_FAMILY=\"#{options[:family]}\""
    )
    else 
      gym(scheme: "scheme")
    end
 end

  desc "Runs all the tests"
  lane :test do
   scan(
	scheme: "test scheme",
	output_types: 'junit',
	custom_report_file_name: 'report.xml'
	)
  end

  lane :iphone do
   
   build(family: "1,2")
   pilot(skip_submission: true)

  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful
    if is_ci?
     slack(
       message: "Successfully uploaded a testversion for #{lane}."
     )
     else 
      puts "Successfully uploaded a testversion for #{lane}"
    end
  end
  error do |lane, exception|
    if is_ci?
     slack(
       message: "error on lane #{lane}: " + exception.message,
      success: false
     )
     else 
      puts "error on lane #{lane}: " + exception.message
    end
  end
end
```

</details>

**Comment by saruye** [comment]
I tested a bit more and the error only comes after building the first lane in our project and after installing cocoapods. I somehow was able to fix it. I think I did `bundle exec gem install cocoapods` or something like it. But I continued to have other weird problems when running fastlane without bundle exec. 
My expectation was that when running fastlane from homebrew I would have less problems with a global messy state of ruby dependencies, but that doesn't seem to be the case. 
I don't know how to give more details about my system, so I guess this ticket can be closed. Right now I am using the gem install of fastlane again and that seems to work.
